polynote-server/src/main/scala/polynote/server/repository/FileBasedRepository.scala

```diff
@@ -265,7 +266,10 @@ class FileBasedRepository(
     fmt            <- NotebookFormat.getFormat(pathOf(path))
     content        <- fs.readPathAsString(pathOf(path))
     (noExtPath, _)  = extractExtension(path)
-    nb             <- fmt.decodeNotebook(noExtPath, content)
+    nb             <- fmt.decodeNotebook(noExtPath, content).catchSome {
+      case err: io.circe.ParsingFailure =>
+        ZIO.fail(new ParsingFailure("Unable to parse file at " + path + ", " + err.toString, err))
```

Comment:
```suggestion
        ZIO.fail(new ParsingFailure(s"Unable to parse file at $path, $err", err))
```
[nit] generally we prefer string interpolation to concatenation in Scala